did/plugins/public_inbox.py

```diff
@@ -143,8 +144,8 @@ def __get_msgs_from_mbox(self, mbox: mailbox.mbox) -> list[Message]:
 
         return msgs
 
-    def __fetch_thread_root(self, initial_msg: Message) -> Message:
-        msg_id = initial_msg.id()
+    def __fetch_thread_root(self, msg: Message) -> typing.Optional[Message]:
```

Comment:
using `msg` instead of `initial_msg` triggers error:

```
did/plugins/public_inbox.py:157:12: R1704: Redefining argument with the local name 'msg' (redefined-argument-from-local)
```

as msg is overridden by:

```python
 for msg in self.__get_msgs_from_mbox(mbox):
            if msg.is_thread_root():
            ...
```

is the overriding something you wanted?

Reply:
You're right. I dropped the commit